BUG: Error No valid VE_LED_TYPE selected when deactivating Status LED in menuconfig

```
/home/joelh/Dev/vigilant-engine/components/vigilant_engine/src/status_led.c: In function 'configure_led':
/home/joelh/Dev/vigilant-engine/components/vigilant_engine/src/status_led.c:150:10: error: #error "No valid VE_LED_TYPE selected"
  150 |         #error "No valid VE_LED_TYPE selected"
      |          ^~~~~
make[2]: *** [esp-idf/vigilant_engine/CMakeFiles/__idf_vigilant_engine.dir/build.make:126: esp-idf/vigilant_engine/CMakeFiles/__idf_vigilant_engine.dir/src/status_led.c.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/joelh/Dev/vigilant-engine/components/vigilant_engine/src/http_server.c:332:13: warning: 'disconnect_handler' defined but not used [-Wunused-function]
  332 | static void disconnect_handler(void* arg, esp_event_base_t event_base,
      |             ^~~~~~~~~~~~~~~~~~
make[1]: *** [CMakeFiles/Makefile2:6514: esp-idf/vigilant_engine/CMakeFiles/__idf_vigilant_engine.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
make failed with exit code 2, output of the command is in the /home/joelh/Dev/vigilant-engine/build/log/idf_py_stderr_output_48518 and /home/joelh/Dev/vigilant-engine/build/log/idf_py_stdout_output_48518

❌ Command failed with exit code 2
(venv) [joelh@joel-zenbook vigilant-engine]$ 
```

This error blocks building and occurs when you have these settings in the menuconfig:
<img width="378" height="110" alt="Image" src="https://github.com/user-attachments/assets/fc0a15c3-0c2f-4679-8302-9e6d485abac2" />

--- a/components/vigilant_engine/src/status_led.c
+++ b/components/vigilant_engine/src/status_led.c
@@ -96,61 +96,64 @@ esp_err_t configure_led()
         led_on = 0;
         led_off = 1;
     }
+    #if ENABLE_LED
+        #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
+            ESP_LOGI(TAG, "Initializing WS2812B status LED");
+            if (s_strip) return ESP_OK; // already initialized
 
-    #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
-        ESP_LOGI(TAG, "Initializing WS2812B status LED");
-        if (s_strip) return ESP_OK; // already initialized
+            led_strip_config_t strip_cfg = {
+                .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
+                .max_leds = 1,
+                .led_model = LED_MODEL_WS2812,
+                .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
+                .flags = {
+                    .invert_out = INVERT_LED,
+                }
+            };
 
-        led_strip_config_t strip_cfg = {
-            .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
-            .max_leds = 1,
-            .led_model = LED_MODEL_WS2812,
-            .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
-            .flags = {
-                .invert_out = INVERT_LED,
+            led_strip_rmt_config_t rmt_cfg = {
+                .clk_src = RMT_CLK_SRC_DEFAULT,
+                .resolution_hz = (10 * 1000 * 1000),
+                .mem_block_symbols = 64,
+                .flags = {
+                    .with_dma = false,
+                }
+            };
+
+            esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
+            if (err != ESP_OK) {
+                ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
+                s_strip = NULL;
+                return err;
             }
-        };
 
-        led_strip_rmt_config_t rmt_cfg = {
-            .clk_src = RMT_CLK_SRC_DEFAULT,
-            .resolution_hz = (10 * 1000 * 1000),
-            .mem_block_symbols = 64,
-            .flags = {
-                .with_dma = false,
+            esp_err_t off_err = ws2812b_status_led_off();
+            if (off_err == ESP_OK) {
+                ESP_LOGI(TAG, "Done initializing status LED");
             }
-        };
-
-        esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
-            s_strip = NULL;
-            return err;
-        }
-
-        esp_err_t off_err = ws2812b_status_led_off();
-        if (off_err == ESP_OK) {
-            ESP_LOGI(TAG, "Done initializing status LED");
-        }
-        return off_err;
-    #elif defined(CONFIG_VE_LED_TYPE_NONE) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
-        #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
-        
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
-        
-        #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
+            return off_err;
+        #elif defined(CONFIG_VE_LED_TYPE_NONE) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
+            #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
+            
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
+            
+            #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
+            #endif
+        #else
+            #error "No valid VE_LED_TYPE selected"
+            ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
         #endif
+        ESP_LOGI(TAG, "Done initializing status LED");
     #else
-        #error "No valid VE_LED_TYPE selected"
-        ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
+        ESP_LOGI(TAG, "ENABLE_LED is false, skipping status LED initialization");
     #endif
-    ESP_LOGI(TAG, "Done initializing status LED");
     return ESP_OK;
 }
 

--- a/components/vigilant_engine/src/status_led.c
+++ b/components/vigilant_engine/src/status_led.c
@@ -96,64 +96,61 @@ esp_err_t configure_led()
         led_on = 0;
         led_off = 1;
     }
-    #if ENABLE_LED
-        #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
-            ESP_LOGI(TAG, "Initializing WS2812B status LED");
-            if (s_strip) return ESP_OK; // already initialized
+    #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
+        ESP_LOGI(TAG, "Initializing WS2812B status LED");
+        if (s_strip) return ESP_OK; // already initialized
 
-            led_strip_config_t strip_cfg = {
-                .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
-                .max_leds = 1,
-                .led_model = LED_MODEL_WS2812,
-                .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
-                .flags = {
-                    .invert_out = INVERT_LED,
-                }
-            };
-
-            led_strip_rmt_config_t rmt_cfg = {
-                .clk_src = RMT_CLK_SRC_DEFAULT,
-                .resolution_hz = (10 * 1000 * 1000),
-                .mem_block_symbols = 64,
-                .flags = {
-                    .with_dma = false,
-                }
-            };
-
-            esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
-            if (err != ESP_OK) {
-                ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
-                s_strip = NULL;
-                return err;
+        led_strip_config_t strip_cfg = {
+            .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
+            .max_leds = 1,
+            .led_model = LED_MODEL_WS2812,
+            .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
+            .flags = {
+                .invert_out = INVERT_LED,
             }
+        };
 
-            esp_err_t off_err = ws2812b_status_led_off();
-            if (off_err == ESP_OK) {
-                ESP_LOGI(TAG, "Done initializing status LED");
+        led_strip_rmt_config_t rmt_cfg = {
+            .clk_src = RMT_CLK_SRC_DEFAULT,
+            .resolution_hz = (10 * 1000 * 1000),
+            .mem_block_symbols = 64,
+            .flags = {
+                .with_dma = false,
             }
-            return off_err;
-        #elif defined(CONFIG_VE_LED_TYPE_NONE) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
-            #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
-                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
-                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
-                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
+        };
+
+        esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
+            s_strip = NULL;
+            return err;
+        }
+
+        esp_err_t off_err = ws2812b_status_led_off();
+        if (off_err == ESP_OK) {
+            ESP_LOGI(TAG, "Done initializing status LED");
+        }
+        return off_err;
+    #elif defined(CONFIG_VE_LED_TYPE_GENERIC) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
+        #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
+            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
+            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
+            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
             
-                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
-                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
-                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
+            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
+            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
+            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
             
-            #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
-                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
-                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
-            #endif
-        #else
-            #error "No valid VE_LED_TYPE selected"
-            ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
+        #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
+            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
+            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
         #endif
-        ESP_LOGI(TAG, "Done initializing status LED");
     #else
-        ESP_LOGI(TAG, "ENABLE_LED is false, skipping status LED initialization");
+        #error "No valid VE_LED_TYPE selected"
+        ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
     #endif
+    ESP_LOGI(TAG, "Done initializing status LED");
+
     return ESP_OK;
 }
 

--- a/components/vigilant_engine/src/status_led.c
+++ b/components/vigilant_engine/src/status_led.c
@@ -96,61 +96,64 @@ esp_err_t configure_led()
         led_on = 0;
         led_off = 1;
     }
-    #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
-        ESP_LOGI(TAG, "Initializing WS2812B status LED");
-        if (s_strip) return ESP_OK; // already initialized
+    #if ENABLE_LED
+        #if defined(CONFIG_VE_LED_TYPE_WS2812B) // Initialization for WS2812B LED strip using RMT peripheral
+            ESP_LOGI(TAG, "Initializing WS2812B status LED");
+            if (s_strip) return ESP_OK; // already initialized
 
-        led_strip_config_t strip_cfg = {
-            .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
-            .max_leds = 1,
-            .led_model = LED_MODEL_WS2812,
-            .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
-            .flags = {
-                .invert_out = INVERT_LED,
+            led_strip_config_t strip_cfg = {
+                .strip_gpio_num = CONFIG_VE_STATUS_WS2812B_PIN,
+                .max_leds = 1,
+                .led_model = LED_MODEL_WS2812,
+                .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,
+                .flags = {
+                    .invert_out = INVERT_LED,
+                }
+            };
+
+            led_strip_rmt_config_t rmt_cfg = {
+                .clk_src = RMT_CLK_SRC_DEFAULT,
+                .resolution_hz = (10 * 1000 * 1000),
+                .mem_block_symbols = 64,
+                .flags = {
+                    .with_dma = false,
+                }
+            };
+
+            esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
+            if (err != ESP_OK) {
+                ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
+                s_strip = NULL;
+                return err;
             }
-        };
 
-        led_strip_rmt_config_t rmt_cfg = {
-            .clk_src = RMT_CLK_SRC_DEFAULT,
-            .resolution_hz = (10 * 1000 * 1000),
-            .mem_block_symbols = 64,
-            .flags = {
-                .with_dma = false,
+            esp_err_t off_err = ws2812b_status_led_off();
+            if (off_err == ESP_OK) {
+                ESP_LOGI(TAG, "Done initializing status LED");
             }
-        };
-
-        esp_err_t err = led_strip_new_rmt_device(&strip_cfg, &rmt_cfg, &s_strip);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "led_strip_new_rmt_device failed: %s", esp_err_to_name(err));
-            s_strip = NULL;
-            return err;
-        }
-
-        esp_err_t off_err = ws2812b_status_led_off();
-        if (off_err == ESP_OK) {
-            ESP_LOGI(TAG, "Done initializing status LED");
-        }
-        return off_err;
-    #elif defined(CONFIG_VE_LED_TYPE_GENERIC) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
-        #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
+            return off_err;
+        #elif defined(CONFIG_VE_LED_TYPE_GENERIC) // Applies initialization for both RGB and BLINK modes, as they both use GPIO output
+            #if defined(CONFIG_VE_STATUS_LED_MODE_RGB)
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_RED);
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_GREEN);
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLUE);
             
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_RED, GPIO_MODE_OUTPUT);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_GREEN, GPIO_MODE_OUTPUT);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLUE, GPIO_MODE_OUTPUT);
             
-        #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
-            gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
-            gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
+            #elif defined(CONFIG_VE_STATUS_LED_MODE_BLINK)
+                gpio_reset_pin(CONFIG_VE_STATUS_LED_GPIO_BLINK);
+                gpio_set_direction(CONFIG_VE_STATUS_LED_GPIO_BLINK, GPIO_MODE_OUTPUT);
+            #endif
+        #else
+            #error "No valid VE_LED_TYPE selected"
+            ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
         #endif
+        ESP_LOGI(TAG, "Done initializing status LED");
     #else
-        #error "No valid VE_LED_TYPE selected"
-        ESP_LOGI(TAG, "NO VALID VE_LED_TYPE SELECTED, NOT INITIALIZING STATUS LED");
+        ESP_LOGI(TAG, "ENABLE_LED is false, skipping status LED initialization");
     #endif
-    ESP_LOGI(TAG, "Done initializing status LED");
-
     return ESP_OK;
 }
 

